app/services/create_github_repo_service.rb

```diff
@@ -0,0 +1,208 @@
+  #         with github_repository id and relay id.
+  #
+  # github_repository - GitHubRepository object of the newly created repo.
+  #
+  # Returns the created AssignmentRepo/GroupAssignmentRepo object
+  # or raises a Result::Error on failure
+  def create_assignment_repo!(github_repository)
+    assignment_repo_attrs = {
+      github_repo_id: github_repository.id,
+      github_global_relay_id: github_repository.node_id
+    }
+    assignment_repo_attrs[entity.humanize] = entity.collaborator
+    assignment_repo = entity.repos.build(assignment_repo_attrs)
+    assignment_repo.save!
+    assignment_repo
+  rescue ActiveRecord::RecordInvalid => error
+    raise Result::Error.new errors(:default), error.message
+  end
+
+  def delete_github_repository(github_repo_id)
+    return true if github_repo_id.nil?
+    organization.github_organization.delete_repository(github_repo_id)
+  rescue GitHub::Error
+    true
+  end
+
+  # Public: Push starter code to the newly created GitHub
+  #         repository.
+  #
+  # assignment_repository - GitHubRepository in which starter code is to be imported.
+  #
+  # Returns true of raises a Result::Error.
+  def push_starter_code!(assignment_repository)
+    client = assignment.creator.github_client
+    starter_code_repo_id = assignment.starter_code_repo_id
+    starter_code_repository = GitHubRepository.new(client, starter_code_repo_id)
+
+    assignment_repository.get_starter_code_from(starter_code_repository)
+  rescue GitHub::Error => error
+    raise Result::Error.new errors(:starter_code_import_failed), error.message
+  end
+
+  # Public: Ensure that we can make a private repository on GitHub.
+  #
+  # Returns True or raises a Result::Error with a helpful message.
+  # rubocop:disable Metrics/AbcSize
+  def verify_organization_has_private_repos_available!
+    return true if assignment.public?
+
+    begin
+      github_organization_plan = GitHubOrganization.new(organization.github_client, organization.github_id).plan
+    rescue GitHub::Error => error
+      raise Result::Error, error.message
+    end
+
+    owned_private_repos = github_organization_plan[:owned_private_repos]
+    private_repos       = github_organization_plan[:private_repos]
+
+    return true if owned_private_repos < private_repos
+    raise Result::Error, errors(:private_repos_not_available, github_organization_plan)
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  # Public: Add user/team to the GitHubRepository based on the type of assignment.
+  #         Calls #add_user_to_github_repository! if it is an Assignment.
+  #         Calls #add_group_to_github_repository! if it is a GroupAssignment.
+  #
+  # github_repository - GitHubRepository in which we need to add collaborator.
+  #
+  # Returns true if collaborator added or raises a Result::Error.
+  def add_collaborator_to_github_repository!(github_repository)
+    send("add_#{entity.humanize}_to_github_repository!", github_repository)
```

Comment:
Can we write this without `send` and string interpolation? This makes for code that's difficult to read and search for. Explicit conditionals are preferred.